### PR TITLE
Test adding a normal cv to composite-cv

### LINKED ISSUE #772: Test adding a a normal cv to composite-cv
a) create a normal content-view
b) associate repo to above CV
c) publish the above CV to get version1
d) fetch the version1_id
e) Now create a composite content-view , let's call this com-CV
f) associate version version1_id to com-CV
g) check for component id, under components matches to that of the version1_id which you just associated in the previous step.

### PR DESCRIPTION
**Pull request by oshtaier**
Added API test case to Resolve #772

### CODE REVIEW

**Comment by oshtaier** [comment]
nosetests tests/foreman/api/test_contentview.py -m test_add_normal_cv_to_composite
.
----------------------------------------------------------------------
Ran 1 test in 78.074s

OK

File: tests/foreman/api/test_contentview.py [hunk 1]
```diff
@@ -428,6 +428,48 @@ def test_positive_promote_3(self):
...
+
+        @Assert: Content view can be created and assigned to
+        composite one through content view versions
+        mechanism
+
+        @Feature: ContentView
+
+        """
+        content_view = entities.ContentView(organization=self.org.id)
+        content_view.id = content_view.create_json()['id']
```
Comment (Ichimonji10) [inline tests/foreman/api/test_contentview.py]: As of [NailGun 0.7.0](https://github.com/SatelliteQE/nailgun/releases/tag/0.7.0), you can now write this instead:

    content_view = entities.ContentView(organization=self.org.id).create()

Here's a capture from my local system that further illustrates what's going on:

```python
>>> from robottelo import entities
>>> content_view = entities.ContentView(organization=1).create()
>>> type(content_view) == entities.ContentView
True
>>> content_view.id
1043
>>> content_view.get_values()
{'description': None, 'repository': [], 'composite': None, 'component': [], 'label': u'bed9f251-953f-4670-b973-150536e241a3', 'organization': <robottelo.entities.Organization object at 0x7f51ff0ffe90>, 'id': 1043, 'name': u'\u4718\U000269f5'}
>>> content_view.organization.id
1
```

File: tests/foreman/api/test_contentview.py [hunk 2]
```diff
@@ -428,6 +428,48 @@ def test_positive_promote_3(self):
...
+                for version
+                in content_view.read_json()['versions']):
+            entities.ContentViewVersion(id=cvv_id)
+
+        composite_cv = entities.ContentView(
+            id=entities.ContentView(
+                organization=self.org.id,
+                composite=True,
+                ).create_json()['id']
+        )
```
Comment (Ichimonji10) [inline tests/foreman/api/test_contentview.py]: You can write this instead:

```python
composite_cv = entities.ContentView(
    composite=True,
    organization=self.org.id,
).create()
```

File: tests/foreman/api/test_contentview.py [hunk 3]
```diff
@@ -428,6 +428,48 @@ def test_positive_promote_3(self):
...
+        """
+        content_view = entities.ContentView(organization=self.org.id)
+        content_view.id = content_view.create_json()['id']
+        content_view.set_repository_ids([self.yum_repo.id])
+        content_view.publish()
+        for cvv_id in (
+                version['id']
+                for version
+                in content_view.read_json()['versions']):
+            entities.ContentViewVersion(id=cvv_id)
```
Comment (Ichimonji10) [inline tests/foreman/api/test_contentview.py]: Creating a `ContentViewVersion` object and doing nothing with that object is a no-op. Can this be simplified?

File: tests/foreman/api/test_contentview.py [hunk 4]
```diff
@@ -428,6 +428,48 @@ def test_positive_promote_3(self):
...
+        composite_cv = entities.ContentView(
+            id=entities.ContentView(
+                organization=self.org.id,
+                composite=True,
+                ).create_json()['id']
+        )
+
+        client.put(
+            composite_cv.path(),
+            {'content_view': {'component_ids': [cvv_id]}},
```
Comment (Ichimonji10) [inline tests/foreman/api/test_contentview.py]: Earlier in this test, I see that `cvv_id` is being used as the assigned-to variable in a loop. How many content view IDs are there? Does it matter which content view ID is usd in this PUT request, and are you sure that the loop will iterate through available values and assign the right value to `cvv_id`?

File: tests/foreman/api/test_contentview.py [hunk 5]
```diff
@@ -428,6 +428,48 @@ def test_positive_promote_3(self):
...
+            auth=get_server_credentials(),
+            verify=False,
+        ).raise_for_status()
+
+        cv_attrs = composite_cv.read_json()
+        self.assertEqual(cvv_id, cv_attrs['component_ids'][0])
+        self.assertEqual(
+            content_view.id,
+            cv_attrs['components'][0]['content_view_id'],
+            )
```
Comment (Ichimonji10) [inline tests/foreman/api/test_contentview.py]: Un-indent this closing parentheses.

**Comment by oshtaier** [comment]
Thanks for comments! Fixed all the issues.
Verified both through API calls and UI, step by step, to make sure that all necessary requirements were met and logic works in a proper way.

**Comment by Ichimonji10** [comment]
Nice job. ACK.

```
$ nosetests tests/foreman/api/test_contentview.py -m test_add_normal_cv_to_composite
.
----------------------------------------------------------------------
Ran 1 test in 55.776s

OK
```

**Comment by omaciel** [comment]
ACK

**Pull request merged**